pkg/cloudprovider/providers/vsphere/vsphere_util.go

```diff
@@ -293,3 +297,108 @@ func (vs *VSphere) cleanUpDummyVMs(dummyVMPrefix string) {
 		}
 	}
 }
+
+func setNodeDisk(
+	nodeVolumeMap map[types.NodeName]map[string]bool,
+	volumePath string,
+	nodeName types.NodeName,
+	check bool) {
+
+	volumeMap := nodeVolumeMap[nodeName]
+
+	if volumeMap == nil {
+		volumeMap = make(map[string]bool)
+		nodeVolumeMap[nodeName] = volumeMap
+	}
+	volumeMap[volumePath] = check
+}
+
+func createDummyVirtualDisk(ctx context.Context, ds *vclib.Datastore, dsFolder string) (string, error) {
+	dsFolderPath := filepath.Clean(ds.Path(dsFolder)) + "/"
+	dummyDiskVolPath := dsFolderPath + DummyDiskName
+	dummyDiskVolOptions := vclib.VolumeOptions{
+		SCSIControllerType: vclib.LSILogicControllerType,
+		DiskFormat:         "thin",
+		CapacityKB:         20480,
+	}
+	dummyDisk := diskmanagers.VirtualDisk{
+		DiskPath:      dummyDiskVolPath,
+		VolumeOptions: &dummyDiskVolOptions,
+		VMOptions:     &vclib.VMOptions{},
+	}
+	dummyDiskCanonicalVolPath, err := dummyDisk.Create(ctx, ds)
+	if err != nil {
+		glog.Warningf("Failed to create a dummy vsphere volume with volumeOptions: %+v on datastore: %+v. err: %+v", dummyDiskVolOptions, ds, err)
+		return "", err
+	}
+	return dummyDiskCanonicalVolPath, nil
+}
+
+func deleteDummyVirtualDisk(ctx context.Context, ds *vclib.Datastore) error {
+	kubeVolsPath := filepath.Clean(ds.Path(VolDir)) + "/"
+	dummyDiskVolPath := kubeVolsPath + "kube-dummyDisk.vmdk"
+	disk := diskmanagers.VirtualDisk{
+		DiskPath:      dummyDiskVolPath,
+		VolumeOptions: &vclib.VolumeOptions{},
+		VMOptions:     &vclib.VMOptions{},
+	}
+	err := disk.Delete(ctx, ds)
+	if err != nil {
+		glog.Errorf("Failed to delete dummy vsphere volume with dummyDiskVolPath: %s. err: %+v", dummyDiskVolPath, err)
+		return err
+	}
+	return nil
+}
+
```

Comment:
It will be helpful if you explain the logic in brief in comment.
For reader it will be easy to follow code.

Reply:
I have added comments for the underlying functions and at certain places where it might be difficult to understand.